apps/bmd/src/AppRoot.tsx

```diff
@@ -371,18 +375,23 @@ class AppRoot extends React.Component<Props, State> {
 
   public resetTally = () => {
     this.setState(
-      ({ election }) => ({
+      ({ electionDefinition }) => ({
         ballotsPrintedCount: this.initialAppState.ballotsPrintedCount,
-        tally: getZeroTally(election!),
+        tally: getZeroTally(electionDefinition?.election!),
       }),
       this.storeAppState
     )
   }
 
   public updateTally = () => {
     this.setState(
-      ({ ballotsPrintedCount, election: e, tally: prevTally, votes }) => {
-        const election = e!
+      ({
+        ballotsPrintedCount,
+        electionDefinition: e,
+        tally: prevTally,
+        votes,
+      }) => {
+        const election = e?.election!
```

Comment:
Using both null chaining and non-null assertions together is a little weird. Maybe just do `e!.election`? Then, at least it'll crash here if it's going to crash.

This construct is in several other places in this PR.